controllers/edgedevice_controller.go

```diff
@@ -78,6 +81,18 @@ func (r *EdgeDeviceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	if edgeDevice.DeletionTimestamp != nil {
+		edsr, err := r.EdgeDeviceSignedRequestRepository.Read(ctx, edgeDevice.Name, r.InitialDeviceNamespace)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return ctrl.Result{}, nil
+			}
+			return ctrl.Result{Requeue: true}, err
```

Comment:
the `Requeue: true` will not be considered if an error occurred and can be skipped:

https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/internal/controller/controller.go#L322